schemas/didDocument/didDocument.json

```diff
@@ -0,0 +1,40 @@
+{
+    "$id": "/did-core.didDocument",
+    "type": "object",
+    "properties": {
+      "@context": {
+        "title": "@context",
+        "description": "https://w3c.github.io/did-core-registry/#context",
+        "anyOf": [
+          { "type": "string" },
+          { "type": "array",
+            "items": [
+              {
+                "type": "string",
+                "const": "https://www.w3.org/ns/did/v1"
+              }
+            ] 
+          },
+          { "type": "object" }
```

Comment:
```suggestion
```

Reply:
Does this not break support for embedding contexts, isn't that a feature people want?

Reply:
Yes, it does break support for embedding contexts. It's a side effect of the grand compromise decision made at the F2F (I'm pretty sure almost no one knew that this was one of the side effects, but we didn't get to talk about it because we didn't get to talk about the second proposal at any depth). 

I'm personally fine w/ that decision as I don't think it harms things other than not allowing arbitrary extensibility, which we probably shouldn't support. If people do want to do arbitrary extensibility w/ JSON-LD, they can (and it's safe to do so as long as you use the JSON-LD tooling)... the spec states that you won't be "globally interoperable"... you'll just be interoperable with the JSON-LD folks.

Hmm... which raises the question on whether or not `@context` should be allowed at all because the spec says that, for JSON-only versions, `@context` is forbidden.

So, what I think we actually need is a JSON Schema for JSON-only mode and another JSON Schema for JSON-LD mode. The JSON-only mode should forbid the use of `@context`, the JSON-LD one should have the rules defined above... and then we can debate if object should be allowed. :(

/cc @jricher so he's aware of the complexity that the JSON-only, don't put `@context` in JSON-only documents, rule is creating. It's creating non-trivial branching conditions that we didn't have before.

One way that we could fix this so we use the same JSON Schema for both JSON-only and JSON-LD is to say: You are in JSON-only mode if you get a application/did+json media type - and `@context` is allowed but MUST be ignored, you are in JSON-LD mode if you get an application/did+ld+json media type and `@context` is allowed and MUST be processed.

Reply:
The simplest approach is to make 2 did document schema's one for linked data and one for "pure json". 

"pure json" documents should NEVER have an `@context`... because that was the whole point of the F2F grand compromise.... 

If software sees an `@context` it should be able to assume linked data... if it does not... it should assume that the document is "pure json"... 

otherwise, we should just scrap this registry and go back to making `@context` mandatory... if json only representations are indistinguishable from json-ld representations we will be right back where we started.

Reply:
> The simplest approach is to make 2 did document schema's one for linked data and one for "pure json".

This makes me sad, but it's fine for now, more below...

> "pure json" documents should NEVER have an @context... because that was the whole point of the F2F grand compromise.

Well, that was beside the point wrt. the grand compromise. I know you wanted this, and a few others wanted this... I think it's a misguided reason because - one of the things that motivated this "no `@context` in JSON-only" argument is that you should do content sniffing on the data to figure out if you have a JSON-only document or a JSON-LD document. Content sniffing is generally considered to be a practice of last resort (and there are plenty of scenarios where it blows up in your face or becomes horribly complicated).

We just merged a PR from @jricher (where we [had a conversation](https://github.com/w3c/did-core/pull/197/files#r383531714)) that contemplated making the media type the deciding factor in telling you whether or not you have a JSON-only or JSON-LD document. If media type becomes the only mechanism, then in JSON-only you can have `@context`, but you need to ignore it for application/did+json. For application/did+ld+json, you must process `@context`.

> If software sees an @context it should be able to assume linked data... if it does not... it should assume that the document is "pure json"...

This is an argument for content sniffing which is a generally reviled practice and is only suggested as an option when developers screw things up in the wild and implementations, in an attempt to stop the ecosystem from breaking, implement content sniffing. You almost never want to start a standard out by adding content sniffing heuristics to the standard.

> otherwise, we should just scrap this registry and go back to making @context mandatory... if json only representations are indistinguishable from json-ld representations we will be right back where we started.

Well, we don't have to do that. The question is how are they distinguishable... through the media type (good and clean), or via content sniffing (Terrible McHorrible Solution of Last Resort).

Reply:
I'm so confused... what is the difference between a json only document with an `@context` and ` application/did+json` media type and a json-ld document with a context you can ignore processing... 

Thats exactly the same thing we had before the F2F.... 

I'm all for being explicit with media type, but it's not a solution to the problem. The problem is given a did document (not resolver metadata), how do I tell if the document is "pure json"... If we can't tell without resolver metadata, and the resolver stuff is not part of this WG, we can't actually solve this in any meaningful way.

Content sniffing is bad, but so is not allowing a method implementer to express their representation without relying on metadata... if the method returns a document without an `@context` (totally valid, its optional)... they are intending to not be linked data... we know thats what they want, because we have just had the F2F and @csuwildcat @selfissued @mikelodder7 and others have clearly stated their intention to potentially return documents without an `@context` now that it is allowed... we should use media type when we can, but we should also not hide from people the implications of the face to face....  `@context` is required for JSON-LD, it follows that its absence implies "Pure JSON"... we won't need to sniff if there is metadata, but we should still be able to sniff without access to metadata.

I suggest we do the following:

1. JSON Schema's for Linked Data and Pure JSON (only difference being `@context` requirement)
2. Use media type whenever possible instead of 1.

Reply:
> we won't need to sniff if there is metadata, but we should still be able to sniff without access to metadata.

Let's see what @jricher thinks about this... and @dlongley... all I'm doing here is raising the concern that I don't think we have consensus on this yet. As long as we're consistent and it doesn't blow up the ecosystem, I'm fine. We have multiple choices in front of us... the group needs to pick a consistent position on this.

Reply:
Just coming in to this conversation (post-merge, it seems), the goal of the new producer/consumer language was in fact to disallow a JSON producer from including a context field. However, the requirement is also that the document type needs to be indicated outside of the document itself. We can't rely on internal data tags (like the presence/absence of a field) because we're going to have completely different encodings that will have different underlying parsers, let alone processors. That's why I raised the metadata issue w3c/did-core#203